OptimizelySDK/Odp/OdpSegmentApiManager.cs

```diff
@@ -0,0 +1,260 @@
+        /// <returns>JSON response from ODP</returns>
+        private string QuerySegments(string apiKey, string endpoint,
+            string query
+        )
+        {
+            HttpResponseMessage response;
+            try
+            {
+                response = QuerySegmentsAsync(apiKey, endpoint, query).GetAwaiter().GetResult();
+            }
+            catch (Exception ex)
+            {
+                _errorHandler.HandleError(ex);
+                _logger.Log(LogLevel.ERROR, $"{AUDIENCE_FETCH_FAILURE_MESSAGE} (network error)");
+                return default;
+            }
+
+            if (!response.IsSuccessStatusCode)
+            {
+                _logger.Log(LogLevel.ERROR,
+                    $"{AUDIENCE_FETCH_FAILURE_MESSAGE} ({(int)response.StatusCode})");
+                return default;
+            }
+
+            return response.Content.ReadAsStringAsync().Result;
+        }
+
+        /// <summary>
+        /// Asynchronous handler for querying the ODP GraphQL endpoint
+        /// </summary>
+        /// <param name="apiKey">ODP API Key</param>
+        /// <param name="endpoint">Fully-qualified ODP GraphQL endpoint URL</param>
+        /// <param name="query">GraphQL query</param>
+        /// <returns>JSON response from ODP</returns>
+        private async Task<HttpResponseMessage> QuerySegmentsAsync(string apiKey, string endpoint,
+            string query
+        )
+        {
+            var request = BuildRequestMessage(apiKey, endpoint, query);
+
+            return await _httpClient.SendAsync(request);
+        }
+
+        /// <summary>
+        /// Produces the request GraphQL query payload 
+        /// </summary>
+        /// <param name="apiKey">ODP API Key</param>
+        /// <param name="endpoint">Fully-qualified ODP GraphQL endpoint URL</param>
+        /// <param name="query">GraphQL query</param>
+        /// <returns>Formed HTTP request message ready to be transmitted</returns>
+        private static HttpRequestMessage BuildRequestMessage(string apiKey, string endpoint,
+            string query
+        )
+        {
+            var request = new HttpRequestMessage
+            {
+                RequestUri = new Uri(endpoint),
+                Method = HttpMethod.Post,
+                Headers =
+                {
+                    {
+                        Constants.HEADER_API_KEY, apiKey
+                    },
+                },
+                Content = new StringContent(query, Encoding.UTF8, "application/json"),
+            };
+
+            return request;
+        }
+
+        /// <summary>
+        /// Ensure a string has content that can be parsed from JSON to an object
+        /// </summary>
+        /// <param name="jsonToValidate">Value containing possible JSON</param>
+        /// <returns>True if content could be interpreted as JSON else False</returns>
+        private static bool CanBeJsonParsed(string jsonToValidate)
```

Comment:
name doesn't align with the functionality. Don't need this method, can be used as a condition in `ParseSegmentsResponseJson`